sh/cmd.go

```diff
@@ -132,11 +132,14 @@ func run(env map[string]string, stdout, stderr io.Writer, cmd string, args ...st
 	c.Stderr = stderr
 	c.Stdout = stdout
 	c.Stdin = os.Stdin
-	log.Println("exec:", cmd, strings.Join(args, " "))
+
+	var quoted []string; for i:=0; i<len(args); i++ { 
+		quoted=append(quoted,fmt.Sprintf("\"%s\"",args[i]));
```

Comment:
if you use %q, it'll escape quotes in the string, which might make debugging easier. It won't make a perfect reusable command line, but neither would this, if a string happened to contain quotes.
```suggestion
		quoted=append(quoted,fmt.Sprintf("%q",args[i]));
```

Reply:
```go
package main
import "fmt"


func main(){
	args := []string{"something",`"not something"`, `'fbar'`}
	for _,arg := range args{
		fmt.Printf("%q\n",arg)

	}

}
```

```
"something"
"\"not something\""
"'fbar'"
```

Reply:
Nice use of formatting!